src/DotNetWorker/Hosting/ServiceCollectionExtensions.cs

```diff
@@ -51,7 +53,19 @@ public static IFunctionsWorkerApplicationBuilder AddFunctionsWorker(this IServic
             // Internal logging            
             services.AddLogging(logging =>
             {
-                logging.Services.AddSingleton<ILoggerProvider, GrpcFunctionsHostLoggerProvider>();
+                // Register an instance so we can use it directly for diagnostic logging.
+                logging.Services.AddSingleton<GrpcFunctionsHostLoggerProvider>();
+                logging.Services.AddSingleton<ILoggerProvider>(s => s.GetRequiredService<GrpcFunctionsHostLoggerProvider>());
+
+                var listener = new DiagnosticListener("Microsoft.Azure.Functions.Worker");
+                services.AddSingleton<DiagnosticSource>(listener);
```

Comment:
Chatted about this offline; but a little concerned about registering these types here.

Reply:
Latest PR removes this in favor of `IWorkerDiagnostics`